src/main/java/by/triumgroup/recourse/service/impl/CourseServiceImpl.java

```diff
@@ -1,49 +1,95 @@
     }
 
     @Override
     public List<Course> findByStatus(Course.Status status, Pageable pageable) {
-        return wrapJPACall(() -> repository.findByStatusOrderByIdDesc(status, pageable));
+        return wrapJPACall(() -> courseRepository.findByStatusOrderByIdDesc(status, pageable));
     }
 
     @Override
-    public boolean registerStudentToCourse(Integer studentId, Course courseId) {
-//        Set<User> registeredStudents = course.getStudents();
-//        registeredStudents.add(student);
-//        wrapJPACall(() -> repository.save(course))
-        throw new NotImplementedException();
+    public void registerStudentToCourse(Integer studentId, Integer courseId) {
+        Optional<Course> courseOptional = wrapJPACallToOptional(() -> courseRepository.findOne(courseId));
+        Optional<User> user = wrapJPACallToOptional(() -> userRepository.findOne(studentId));
+        List<ErrorMessage> errorMessages = new ArrayList<>();
+        if (!courseOptional.isPresent()) {
+            errorMessages.add(new ErrorMessage("course", "Not found"));
+        }
+        if (!user.isPresent()) {
+            errorMessages.add(new ErrorMessage("user", "Not found"));
+        }
+        if (!errorMessages.isEmpty()) {
+            throw new ServiceNotFoundException(errorMessages);
+        }
+        if (user.get().getRole() != User.Role.STUDENT) {
+            errorMessages.add(new ErrorMessage("user", "User must be a student"));
+        }
+        Course course = courseOptional.get();
+        if (course.getStatus() != Course.Status.REGISTRATION) {
+            errorMessages.add(new ErrorMessage("course", "Course must has registration status"));
+        }
+        if (!errorMessages.isEmpty()) {
+            throw new ServiceBadRequestException(errorMessages);
+        }
+        Set<User> registeredStudents = course.getStudents();
+        registeredStudents.add(user.get());
+        wrapJPACall(() -> courseRepository.save(course));
     }
 
     @Override
-    public boolean removeStudentFromCourse(Integer studentId, Course courseId) {
-        throw new NotImplementedException();
+    public void removeStudentFromCourse(Integer studentId, Integer courseId) {
+        Optional<Course> courseOptional = wrapJPACallToOptional(() -> courseRepository.findOne(courseId));
+        Optional<User> user = wrapJPACallToOptional(() -> userRepository.findOne(studentId));
+        List<ErrorMessage> errorMessages = new ArrayList<>();
+        if (!courseOptional.isPresent()) {
+            errorMessages.add(new ErrorMessage("course", "Not found"));
+        }
+        if (!user.isPresent()) {
+            errorMessages.add(new ErrorMessage("user", "Not found"));
+        }
+        if (!errorMessages.isEmpty()) {
+            throw new ServiceNotFoundException(errorMessages);
+        }
+        Course course = courseOptional.get();
+        if (course.getStatus() == Course.Status.FINISHED) {
```

Comment:
Only a student himself can not unregister from finished course, an admin should be able to remove a student from a course in any case.